miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt

```diff
@@ -14,10 +14,24 @@ internal class MiniAppWebViewClient(
     private val customScheme: String
 ) : WebViewClient() {
 
-    override fun shouldInterceptRequest(
-        view: WebView,
+    override fun shouldInterceptRequest(view: WebView, request: WebResourceRequest): WebResourceResponse? {
+        val interceptedWebRequest = loader.shouldInterceptRequest(request.url)
+
+        interceptMimeType(interceptedWebRequest, request)
+
+        return interceptedWebRequest
+    }
+
+    @VisibleForTesting
+    internal fun interceptMimeType(
+        interceptedWebRequest: WebResourceResponse?,
         request: WebResourceRequest
-    ): WebResourceResponse? = loader.shouldInterceptRequest(request.url)
+    ) {
+        interceptedWebRequest?.let {
```

Comment:
I'm not sure if it is a good idea as it may slow down. Isn't it solvable in the mini-app build setting itself?
@minh-rakuten @corycaywood @Rajesh-Sundaram

Reply:
@Rajesh-Sundaram on Android 9, RakutenCasa has this error
`Failed to load module script: The server responded with a non-JavaScript MIME type of "text/plain". Strict MIME type checking is enforced for module scripts per HTML spec.`

Is there anything miniapp can solve by itself?

Reply:
I don't think there's any way for the mini app to solve this other than not using the `module` type. So they would have to always bundle their JS into a single file instead.

Mini Apps usually shouldn't have too many files, so would this have that much affect on the loading time?

Also, on iOS side we added the MIME types for everything supported by the backend: [see here](https://github.com/rakutentech/ios-miniapp/blob/master/MiniApp/Classes/Display/MimeTypes.swift). Would it make sense to do the same for Android?

Reply:
@corycaywood For now, Android only has a problem with `module` type so I think only MIME type for javascript is enough.
OS 10 and later, android already solves this issue.

Reply:
@corycaywood Would bundling be more natural for a prod-type of mini-app, especially when a javascript is written in a `type=module` format by a developer?
@minh-rakuten this issue happens only for Android 9?

Reply:
This issue happens on Android 9 and below.

Reply:
I think we need to support `type=module`: it's a new feature from ES6 so it provides some benefits from bundling the JS in a single file. One important benefit is that they have deferred loading by default, which means they don't block rendering of the page while the browsing loading/parsing the JS.